src/main/kotlin/org/wa/rceditor/application/viewmodel/MainViewModel.kt

```diff
@@ -1,113 +1,320 @@
+                    container = it
+                    clearData()
+                    directoryLoaded = true
+                    loadRecourceContainer()
+                }
+    }
+
+    fun handleSaveDocumentSelected() {
+        saveResourceContainer()
+    }
+
+    fun handleAppQuit() {
+        exitProcess(-1)
+    }
+
 
     // ------------ Functions -------------- //
 
-    fun contributors(): ObservableList<ContributorItem> {
-        return model.contributors()
+    fun contributors(): ObservableList<StringProperty> {
+        return contributors
     }
 
     fun addContributor(text: String) {
-        model.addContributor(text)
+        contributors.add(SimpleStringProperty(text))
     }
 
-    fun removeContributor(item: ContributorItem) {
-        model.removeContributor(item)
+    fun removeContributor(item: StringProperty) {
+        contributors.remove(item)
     }
 
-    fun relations(): ObservableList<RelationItem> {
-        return model.relations()
+    fun relations(): ObservableList<StringProperty> {
+        return relations
     }
 
     fun addRelation(text: String) {
-        model.addRelation(text)
+        relations.add(SimpleStringProperty(text))
     }
 
-    fun removeRelation(item: RelationItem) {
-        model.removeRelation(item)
+    fun removeRelation(item: StringProperty) {
+        relations.remove(item)
     }
 
     fun sources(): ObservableList<SourceItem> {
-        return model.sources()
+        return sources
     }
 
     fun addSource(identifier: String, language: String, version: String) {
-        model.addSource(identifier, language, version)
+        sources.add(SourceItem(identifier, language, version))
     }
 
     fun removeSource(item: SourceItem) {
-        model.removeSource(item)
+        sources.remove(item)
     }
 
-    fun checkingEntities(): ObservableList<CheckingEntityItem> {
-        return model.checkingEntities()
+    fun checkingEntities(): ObservableList<StringProperty> {
+        return checkingEntities
     }
 
     fun addCheckingEntity(text: String) {
-        model.addCheckingEntity(text)
+        checkingEntities.add(SimpleStringProperty(text))
     }
 
-    fun removeCheckingEntity(item: CheckingEntityItem) {
-        model.removeCheckingEntity(item)
+    fun removeCheckingEntity(item: StringProperty) {
+        checkingEntities.remove(item)
     }
 
     fun projects(): ObservableList<ProjectItem> {
-        return model.projects()
+        return projects
     }
 
     fun addProject(title: String, versification: String, identifier: String, sort: Int, path: String, category: String) {
-        model.addProject(title, versification, identifier, sort, path, category)
+        projects.add(ProjectItem(title, versification, identifier, sort, path, category))
     }
 
     fun removeProject(item: ProjectItem) {
-        model.removeProject(item)
+        projects.remove(item)
     }
 
-    // ------------- Handlers ---------------- //
+    fun loadRecourceContainer() {
```

Comment:
Another naming detail, but `setViewData` might be more accurate here. Someone might think `loadResourceContainer` means this function is actually loading the container from the disk

Reply:
@mbr4477 Good point

Reply:
bindResourceContainer()

And if this isn't actually binding the resource container to properties, it should be.